### PR TITLE
Add cypress tests

### PR DESCRIPTION
**Pull request by juliushaertl**
Implement simple cypress tests for opening and creating a new document for all supported file types.

To be extended over time, just so that we finally have the structure in place.

There is also a few things that should be moved to @nextcloud/cypress like file upload and folder creation.

### CODE REVIEW

**Comment by juliushaertl** [comment]
Turns out, test failures expected due to https://github.com/nextcloud/server/pull/30385 and i was just not on the latest master locally

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        containers: [1, 2, 3, 4]
+        php-versions: [ '7.4' ]
+        databases: [ 'sqlite' ]
+        server-versions: [ 'master' ]
+
+    name: runner ${{ matrix.containers }}
+
+    services:
+      collabora:
+        image: collabora/code:21.11.2.4.1
```
Comment (juliushaertl) [inline .github/workflows/cypress.yml:28]: We need to handle the welcome popup separately with the follow up release during testing.

File: cypress/e2e/open.spec.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+				.should('be.visible')
+				.and('have.class', 'modal-mask')
+				.and('not.have.class', 'icon-loading')
+
+			cy.get('#collaboraframe').iframe().should('exist').as('collaboraframe')
+			cy.get('@collaboraframe').within(() => {
+				cy.get('#loleafletframe').iframe().should('exist').as('loleafletframe')
+			})
+
+			cy.get('@loleafletframe').find('#main-document-content').should('exist')
```
Comment (marcelklehr) [inline cypress/e2e/open.spec.js:94]: Can we use the waitForCollabora collabora command here?

Comment (juliushaertl) [reply cypress/e2e/open.spec.js]: Yep good catch. I replaced all occurrences with the command and cleaned up the unused commands that have been moved to the package already.

**Review by marcelklehr — APPROVED** [review]

**Pull request merged**